docs/baseplate/core.rst

```diff
@@ -1,19 +1,62 @@
-================================
-Diagnostics: Spans and Observers
-================================
+baseplate.core
+==============
 
-Spans
------
+The heart of the Baseplate framework is its diagnostics system. Here's an
+incomplete example of an application built on the framework::
+
+   def do_something(request):
+       request.some_redis_client.ping()
+
+   def make_app(app_config):
+       ... snip ...
+
+       baseplate = Baseplate()
+       baseplate.configure_metrics(metrics_client)
+       baseplate.add_to_context(
+           "some_redis_client", RedisContextFactory(redis_pool))
+
+       ... snip ...
 
-The heart of Baseplate is its diagnostics system.
+When a request is made which routes to the ``do_something`` handler, a
+:py:class:`~baseplate.core.RootSpan` is automatically created. If the incoming
+request has trace headers, the root span will be identical to the child span in
+the upstream service. When we call ``request.some_redis_client.ping()`` in the
+handler, Baseplate will create a child :py:class:`~baseplate.core.Span` object
+to represent the time taken talking to redis.
```

Comment:
thanks for the feedback. i've updated the following paragraph, lemme know what you think.